netrw/rewire/algebraic_connectivity.py

```diff
@@ -50,17 +50,17 @@ def step_rewire(
         if copy_graph:
             G = copy.deepcopy(G)
 
-        if not nx.is_connected(G):
-            raise ValueError(
-                "Disconnected graph. This method is implemented for undirected, connected graphs."
-            )
-
         if nx.is_directed(G) and directed is True:
```

Comment:
If `directed=False` here the function will not coerce to undirected, right? This might be best to include an `else` here and raise ValueError